Sync Aoe_Static with AOEmedia repo

tmotyl opened the pull request.

Commit:
Added Cache-Control: no-cache for non cacheable sites

Commit:
Changed configuration to read xml files aoe_static.xml

Commit:
Fixed typo and removed aoestatic header with value cache in non cached items

Commit:
Fix to show Headers too if config cache is enabled

tmotyl committed:
Fix cache ID conflict with Full Page Cache,
Rename constants for better readability

tmotyl committed:
Raise version number

fbrnc committed:
Added new marker

fbrnc committed:
Removing CUSTOMERNAME and CARTSUMMARYCOUNT from before_body_end block

fbrnc committed:
Fixed missing marker characters

fbrnc committed:
Added some comments

tmotyl committed:
Make cookies available for js when "use httpOnly"

setting is set in configuration/web/session cookie management
Add proper casting for bolean values in configuration

tmotyl committed:
Fix XSS issue with customer name

Escape customer's name before setting cookie

Commit:
used request string without params as identifier

Commit:
MAGE-239: Varnish doesn't purge the GuidaTv page
- added custom url max age functionality

Commit:
MAGE-239: Varnish doesn't purge the GuidaTv page
- removed empty folder adminhtml/template/aoestatic/

Commit:
MAGE-239: Varnish doesn't purge the GuidaTv page
- fixed bug with not set store id

tmotyl committed:
Add possibility to register own markers for cookies and headers

tmotyl committed:
[!!!] add cookies and headers in the postprocess event instead of preprocess

Commit:
removed bogus acl node

Commit:
fixed translation helper name

Commit:
added mass actions block and actions column

Commit:
fixed exception handler in massDeleteAction()

Commit:
fixed mass delete implementation

Commit:
MAGE-471: added check, which skips caching headers if messages should be printed on current page

Commit:
MAGE-471: added check, which skips caching headers if messages should be printed on current page
- reverted commit committed by mistake to the master branch

Commit:
MAGE-471: added slightly modified changes again

Commit:
Merge pull request #3 from AOEmedia/MAGE-471

MAGE-471: added slightly modified changes again

Commit:
added registry-based custom max-age settings for special cases like special products

Commit:
refactored codestyle, updated folder structure to match magentos structure

Commit:
initial merging from Aoe_PurgeStatic (based on Magneto_Varnish and Aoe_AsyncVarnishCache) into Aoe_Static

Commit:
added missing files

Commit:
fixes and phpdoc

Commit:
implemented tag-based purging

Commit:
fixes

Commit:
updated vcl from varnish 2 to 3 and added BAN

Commit:
fixed cache purging

Commit:
fixed X

Commit:
added cache-type invalidation and checks if cache is enabled

Commit:
renamed phone to aoestatic route name

Commit:
added API methods

Commit:
small code cleanup (phpmd)

Commit:
fixed some minor bugs

Commit:
fixed variable name

Commit:
removed blacklist url patterns

Commit:
Added configuration options for
- use_aoe_asynccache
- purgeadapter (multiselect)
- servers (used by varnish purge adapter)

Refactored Aoe_Static_Helper_Data->_getAdapterInstances to instanciate the adapters from configuration instead of all of the existing ones

Added a blackhole adapter that logs instead of doing any actual purging

Commit:
Added license, dummy readme and cleaned up class comments

Commit:
Fixed typos

Commit:
Fixed warning

Commit:
added url purging after a custom max-age is set

Commit:
fixed max-age bug

Commit:
fixed caching bug

Commit:
Work in progress

Commit:
fixed some typos and bugs

Commit:
added more details to mage::log

Commit:
Merge branch 'cookies' into refactored

Commit:
added a default customer cookie scope

Commit:
added a default customer cookie scope

Commit:
fixed cookie scope

bastianccm committed:
fixed issues in modman file with ./

bastianccm committed:
fixed bug with new-product widgets

bastianccm committed:
added store-level to tag-based caching

LeeSaferite committed:
Fix scope portion of cookie name for customer scope

Commit:
MAGE-726: Fix tests
- added varnish cache hole punching for product.info.addtocart block

Zyava committed:
Fixed MAGE-784: No message during recension save is shown on product detail page
- fixed approach used to check number of pending messages

fbrnc committed:
Merge pull request #4 from AOEpeople/master

Fixed MAGE-784: No message during recension save is shown on product det...

LeeSaferite committed:
Fixing bug with customer login state

LeeSaferite committed:
Code cleanup

LeeSaferite committed:
Remove commented debug code

LeeSaferite committed:
Change how testing 'no-cache' section works

LeeSaferite committed:
Change the ban behaviour to work better with the ban lurker process

LeeSaferite committed:
Update the cookie plugin

LeeSaferite committed:
Simplify tag banning to a single call per server by using a regex pattern

LeeSaferite committed:
Add missing logged in flag cookie

LeeSaferite committed:
Change delimiter to a space

LeeSaferite committed:
Switch to X-Tags as the header for storage and BAN

LeeSaferite committed:
Bump version

bastianccm committed:
added missing constant and fixed 2 missing ; in js

Zyava committed:
replaced unset with remove (varnish vcl) because unset is deprecated

LeeSaferite committed:
Prevent pipe from handling multiple requests

LeeSaferite committed:
Move where info header is added

LeeSaferite committed:
Move BAN/PURGE to top of recv processing

LeeSaferite committed:
Update X-Forwarded-For header only on initial processing

LeeSaferite committed:
Move location of static file request cookie removal

LeeSaferite committed:
Comment cleanup

Zyava committed:
added support for old format - comma-separated list of servers

bastianccm committed:
added asynccache for tags

bastianccm committed:
Merge branch 'refactored' of github.com:AOEmedia/Aoe_Static into refactored

LeeSaferite committed:
Refactor VCL and supporting code

LeeSaferite committed:
Bump version

LeeSaferite committed:
Change where the async cache job is created

LeeSaferite committed:
Bump version

LeeSaferite committed:
Fixing bug with URL banning via regex

LeeSaferite committed:
Bump version

LeeSaferite committed:
Adding proxy header handling for client.ip and ACL checking

LeeSaferite committed:
Split backends and ACLs into different files

Commit:
Apply configuration only if request is dispatched

LeeSaferite committed:
Adding missing PHPDoc

LeeSaferite committed:
Change how layout handles are added

LeeSaferite committed:
Minor refactor of observer

zohmi committed:
added basic composer.json

bastianccm committed:
fixed identifier in anonymous function

bastianccm committed:
added secure controller action

bastianccm committed:
added message retriever, extended js event

bastianccm committed:
refactored logic and cleaned up code

bastianccm committed:
always call home to get newest messages

bastianccm committed:
fixed possible session init error in magento EE 1.14.1.0 on multi-server setups

bastianccm committed:
added localStorage cache to speed up page-render

WinstonN commented on code/Model/Observer.php:
```diff
@@ -4,68 +4,283 @@
+        $session      = Mage::getSingleton('customer/session');
+        /* @var $session Mage_Customer_Model_Session */
+        if ($session->isLoggedIn()) {
+            $loggedIn     = '0';
```
Why is this 0? Shouldn't it be 1?

Commit:
[ENH] Implement event listener for changelog events

Listen to the changelog events created on

* full reindexing
* partial product reindexing (stock, price, etc)
* partial category reindexing

Commit:
Merge pull request #6 from afirlejczyk/refactored

Apply configuration only if request is dispatched

tmotyl committed:
Fix recently viewed products
(works if you have an ajax call on product single view page)

bastianccm committed:
added new event trigger

tmotyl committed:
Adds tagging of the configurable product single view page with child products
tags.

tmotyl committed:
Clear Varnish cache when simple product goes out of stock

Listen to the inventory change event and purge product and category
tags when product status has changed.

tmotyl committed:
Add backend setting which enables verbose logging.

bastianccm committed:
added end-of-live readme

bastianccm committed:
Merge pull request #10 from macopedia/fixForRecentlyViewed

Fix recently viewed products

bastianccm committed:
Merge pull request #8 from macopedia/tagProductPageWithSimple

Adds tagging of the configurable product single view page with child …

Commit:
Add additional fields to Cache-Control for uncached contents

Also force no-store and revalidate for uncached contentes as some browsers cache the content locally causing strange side effects.

Commit:
Prevent throwing js errors if saving in localStorage is not allowed

bastianccm committed:
remove SID from indexAction

bastianccm committed:
trimming customer name

bastianccm committed:
fixed login detection

Commit:
fixed === to ==

Commit:
Varnish 4.0 VCL

fbrnc committed:
Merge pull request #13 from kubaceg/refactored

Varnish 4.0 VCL

Commit:
Fixed label after successful purge

fbrnc committed:
Merge pull request #14 from kubaceg/refactored

Fixed label after successful purge

mathiasschopmans committed:
Add bower.json

Since some projects build their front-end assets separately, we will
add this to bower.

LeeSaferite committed:
Merge pull request #9 from macopedia/purgeWhenStockStatusChanges

Cleare varnish cache when simple product goes out of stock

LeeSaferite committed:
Merge pull request #11 from macopedia/configurableLogging

Add backend setting which enables verbose logging.

Commit:
Merge branch 'master' into refactored

Commit:
Refactored Varnish Purge Adapter

Commit:
Introduced SNS purge adapter

Commit:
Added shell script

fbrnc committed:
Merge pull request #16 from AOEpeople/feature/purge_via_sns

Feature/purge via sns

Commit:
Fix typo

Commit:
Added more shell commands

Commit:
Removed old aoemedia.de email addresses

Commit:
Preparing EsiController (WIP)

Commit:
Removed empty file

Commit:
Added ESI block

Commit:
Added support for caching (not tested yet)

Commit:
Added flag that allows skipping rendering ESI placeholder child blocks

Commit:
Merge branch 'feature/esi'

Commit:
Added HTML comments for better insight

mathiasschopmans committed:
[BUGFIX] Rename EsiPlaceholder::getUrl to getEsiUrl

Commit:
Add option to avoid E.T. phone home if block content is available in localStorage

tessig committed:
Add possibility to overwrite xml settings via cache control singleton's disable method

If Aoe_Static_Model_Cache_Control::disable is called during dispatch process,
the caching should be disabled no matter what the configuration says.

tessig committed:
[BUGFIX] Add public method to get enabled flag from cache control

Commit:
Load the current product in call controller

Load the current product in call controller if the calling page provides
a currentProductId.
The feature can be disabled as the feature may not be required in all
cases.

Commit:
Allow to skip usage of localstorage
